### PR TITLE
Forward rank and world size info to Torchbench models when using dynamo runner

### PR DESCRIPTION
**Pull request by xmfan**
Adding support to pass rank and world_size to torchbench model, via its extra_args parameter: https://github.com/pytorch/benchmark/blob/main/torchbenchmark/util/model.py#L83C80-L83C90

This is used for models which distribute over multiple GPUs e.g. simple_gpt https://github.com/pytorch/benchmark/pull/1867

Also add an option to skip multiprocess only gpu models

Testing via `python benchmarks/dynamo/torchbench.py -d cuda --output=benchmark_logs/performance.csv --inference --performance --timing --print-memory --multiprocess --only simple_gpt`

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @anijain2305

### CODE REVIEW

**Review by Chillee — APPROVED** [review]

**Comment by xmfan** [comment]
Failures about unhandled extra_args should be resolved once the changes from torchbenchmark/util/extra_args.py https://github.com/pytorch/benchmark/pull/1867/files#diff-6ccf656b90a64ee9ee4e55aec794320710e717b65271baeae74e69940524bb6a land

File: benchmarks/dynamo/common.py [hunk 1]
```diff
@@ -3392,6 +3392,15 @@ def run(runner, args, original_dir=None):
             else:
                 try:
                     with tqdm(desc="loading model"):
+                        extra_args = []
+                        if hasattr(args, "rank") and hasattr(args, "world_size"):
```
Comment (H-Huang) [inline benchmarks/dynamo/common.py:3396]: I think we also need to update the hash with your torchbench changes on they land in order for the CI to pick it up

https://github.com/pytorch/pytorch/blob/main/.github/ci_commit_pins/torchbench.txt

**Pull request closed without merging**